addons/web_editor/static/src/components/media_dialog/media_dialog.js

```diff
@@ -144,7 +144,8 @@ export class MediaDialog extends Component {
 
     async save() {
         const selectedMedia = this.selectedMedia[this.state.activeTab];
-        if (selectedMedia.length) {
+        const saveSelectedMedia = selectedMedia.length && (this.state.activeTab !== TABS.ICONS.id || selectedMedia[0].initialIconChanged);
+        if (saveSelectedMedia) {
```

Comment:
Ok, thanks for the explanation. I understand that it's better not to run the `save()` code if the icon hasn't changed. We just close the media dialog :+1:. I just find it annoying to have a reference to the icon tab in the media dialog code... Couldn't we have the same behavior for all the tabs of the media dialog ? If I choose the same image as before, we just close the media dialiog (same for videos, documents, ...) ?

Reply:
@Guillaume-gdi I remember that I thought about this when I added the fix :+1: But this can be an improvement for master **[1]**, I don't think it's a good idea to introduce changes for every media selector to fix icons in stable...
E.g. when you upload an image using `MediaDialog` and add it to the DOM, it will be converted to base64 with a `data-original-src` attribute that points to the attachment URL. So if you try to select the same attachment again on `MediaDialog`, you can't use the original src of the old media (`this.props.media`) to check if it's the same image because the base64 data can hold some "imageProcessing" updates from options... which makes it different from the original attachment in media dialog...

What do you think?

**[1]**: A correct way to improve this is to have something like `createElements()` (a `prepareForSave()` method overridden in each selector component to add specific media saving adaptation) this way, the `save()` method will only keep the general code... :+1:

Reply:
Okay, lets do this : keep the actual fix but add a comment to explain that we will make that improvement for every types of media in master :+1:

Reply:
Updated :+1:

Reply:
Thanks! Can you just split the line in to two ? It's too long :)